jsonnet/components/alertmanager-user-workload.libsonnet

```
@@ -0,0 +1,341 @@
+      automountServiceAccountToken: true,
+    },
+
+    // Adding the serving certs annotation causes the serving certs controller
+    // to generate a valid and signed serving certificate and put it in the
+    // specified secret.
+    //
+    // The ClusterIP is explicitly set, as it signifies the
+    // cluster-monitoring-operator, that when reconciling this service the
+    // cluster IP needs to be retained.
+
+    service+: {
+      metadata+: {
+        annotations: {
+          'service.beta.openshift.io/serving-cert-secret-name': 'alertmanager-user-workload-tls',
+        },
+      },
+      spec+: {
+        ports: [
+          {
+            name: 'web',
+            port: 9095,
+            targetPort: 'web',
+          },
+          {
+            name: 'tenancy',
+            port: 9092,
+            targetPort: 'tenancy',
+          },
+          {
+            name: 'metrics',
+            port: 9097,
+            targetPort: 'metrics',
+          },
+        ],
+        type: 'ClusterIP',
+      },
+    },
+
+    // This changes the alertmanager to be scraped with TLS, authN and authZ,
+    // which are not present in kube-prometheus.
+    serviceMonitor+: {
+      spec+: {
+        endpoints: [
+          {
+            port: 'metrics',
+            interval: '30s',
+            scheme: 'https',
+            tlsConfig: {
+              caFile: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
+              serverName: 'alertmanager-' + cfg.name,
+              certFile: '/etc/prometheus/secrets/metrics-client-certs/tls.crt',
+              keyFile: '/etc/prometheus/secrets/metrics-client-certs/tls.key',
+            },
+          },
+        ],
+      },
+    },
+
+
+    // In order for kube-rbac-proxy to perform a TokenReview and
+    // SubjectAccessReview for authN and authZ the alertmanager ServiceAccount
+    // requires the `create` action on both of these.
+
+    clusterRole: {
+      apiVersion: 'rbac.authorization.k8s.io/v1',
+      kind: 'ClusterRole',
+      metadata: {
+        name: 'alertmanager-' + cfg.name,
+      },
+      rules: [
+        {
+          apiGroups: ['authentication.k8s.io'],
+          resources: ['tokenreviews'],
+          verbs: ['create'],
+        },
+        {
+          apiGroups: ['authorization.k8s.io'],
+          resources: ['subjectaccessreviews'],
+          verbs: ['create'],
+        },
+        {
+          // By default authenticated service accounts are assigned to the `restricted` SCC which implies MustRunAsRange.
+          // This is problematic with statefulsets as UIDs (and file permissions) can change if SCCs are elevated.
+          // Instead, this sets the `nonroot` SCC in conjunction with a static fsGroup and runAsUser security context below
+          // to be immune against UID changes.
+          apiGroups: ['security.openshift.io'],
+          resources: ['securitycontextconstraints'],
+          resourceNames: ['nonroot'],
```

This might be needed depending if gets merged successfully.
 
```suggestion
          resourceNames: ['nonroot-v2'],
```